### PR TITLE
examples/ftpd: Add support for choosing address family

### PR DESCRIPTION
**Pull request by wangchen61698**
## Summary
choose the network protocol (ipv4 or ipv6) to bind network in setting ftpd(default ipv6)

## Impact
N/A

## Testing
CI check

before

ap> ftpd_start
Starting the FTP daemon
ap> FTP daemon [91] started
Adding accounts:

    Root account: USER=root PASSWORD=abc123 HOME=(none)
    User account: USER=ftp PASSWORD=(none) HOME=(none)
    User account: USER=anonymous PASSWORD=(none) HOME=(none)

after

ap> ftpd_start -4
Starting the FTP daemon
ap> FTP daemon [58] started
Adding accounts:

    Root account: USER=root PASSWORD=abc123 HOME=(none)
    User account: USER=ftp PASSWORD=(none) HOME=(none)
    User account: USER=anonymous PASSWORD=(none) HOME=(none)

ap> ftpd_start -6
Starting the FTP daemon
ap> FTP daemon [86] started
Adding accounts:

    Root account: USER=root PASSWORD=abc123 HOME=(none)
    User account: USER=ftp PASSWORD=(none) HOME=(none)
    User account: USER=anonymous PASSWORD=(none) HOME=(none)

### CODE REVIEW

File: examples/ftpd/ftpd_main.c [hunk 1]
```diff
@@ -154,35 +154,75 @@ static void ftpd_accounts(FTPD_SESSION handle)
...
+          break;
+        default:
+          badarg = true;
+          break;
+      }
+    }
+
+  if (badarg)
+    {
+      hint();
```
Comment (pkarashchenko) [inline examples/ftpd/ftpd_main.c:207]: Why an't we `hint();` and `goto out;` from `default` case above? I mean, why do we need `badarg` variable?

File: examples/ftpd/ftpd_main.c [hunk 2]
```diff
@@ -154,35 +154,75 @@ static void ftpd_accounts(FTPD_SESSION handle)
...
 
-  /* Open FTPD */
-
 #if ADDR_FAMILY == AF_INET6
-  handle = ftpd_open(CONFIG_EXAMPLES_FTPD_PORT, AF_INET6);
+  family = AF_INET6;
 #else
-  handle = ftpd_open(CONFIG_EXAMPLES_FTPD_PORT, AF_INET);
+  family = AF_INET;
 #endif
```
Comment (pkarashchenko) [inline examples/ftpd/ftpd_main.c:187]: Let's remove this and change variable declaration to `int family = ADDR_FAMILY;`

Comment (pkarashchenko) [reply examples/ftpd/ftpd_main.c:187]: Or to `int family = AF_UNSPEC;` like in `apps/examples/ftpc/ftpc_main.c`

File: examples/ftpd/ftpd_main.c (hunk 1 above)
Comment (wangchen61698) [reply examples/ftpd/ftpd_main.c:207]: The variable is to prompt user that the input is unappropriate to the connmand.

图片

**Review by pkarashchenko — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**